### DISCUSSION TITLE
why does meson requests exe_wrapper?

### DISCUSSION THREAD

**Discussion by stsp** [mesonbuild/meson#12653] [Q&A]
I use host and cross tools in my project.
That works fine on x86_64, but on aarch64
I am getting this:
`ERROR: An exe_wrapper is needed but was not found.`
Needed by whom?
My project uses host tool-chain just as
well as the cross one, and no wrappers
are needed at all.
How to tell meson to just stop doing that?

**Comment by dcbaker** [comment]
That means you're trying to run a binary being built for the host machine, but your host binaries cannot run on your build machine.

The most likely situation you have is either: 1) you're trying to run tests built for the host machine, or 2) you're trying to use a host binary to generate code

**Reply by stsp** [reply]
I actually looked into a meson
sources, and found that it just
compares the `[host_machine]`
settings against the real host
machine. From there I also found
the solution:
```
[properties]
needs_exe_wrapper = false
```

So do we have a meson bug here?
Because I didn't actually try to run
anything! Meson was trying a guess-work
instead, which is very bad.

**Reply by dcbaker** [reply]
You're cross compiling from x86_64 to aarch64? You need an exe_wrapper if you're running host binaries. That setting tells Meson "My host binaries can run unaltered and unemulated on my build machine. Think x86 binaries on an x86_64 host.

What project are you working on?

**Reply by stsp** [reply]
> You're cross compiling from x86_64 to aarch64?

Opposite: from aarch64 to x86_32.
But I never run such binaries.
I am compiling the native aarch64
shared libs, and do not them either.
This project provides only libs, no
runnable executable.
As such, meson should not worry
about an exe_wrapper.

> What project are you working on?

https://github.com/dosemu2/fdpp

**Comment by stsp** [comment]
So do we have a bug here,
or should I just keep using
```
[properties]
needs_exe_wrapper = false
```
?

**Comment by brunvonlope** [comment]
The situation will be less worse in the future. In master, meson finally tells what command needs the exe_wrapper: https://github.com/mesonbuild/meson/commit/e9e582020898e52d17b4c48afe65589bc1dee065
